scripts/install_prereqs.py

```diff
@@ -0,0 +1,453 @@
+        (bool): True if windows, False otherwise.
+    """
+    return sys.platform.startswith('win')
+
+
+def is_linux_os():
+    """Check if the current OS is Linux.
+
+    Returns:
+        (bool): True if windows, False otherwise.
+    """
+    return sys.platform.startswith('linux')
+
+
+def is_mac_os():
+    """Check if the current OS is mac.
+
+    Returns:
+        (bool): True if windows, False otherwise.
+    """
+    return sys.platform.startswith('darwin')
+
+
+def get_system_package_manager():
+    """ Get package manger based on the platform.
+    brew for mac, apt for linux, choco for windows.
+
+    Raises:
+        ValueError: If the current OS could not be identified.
+
+    Returns:
+        (str): Package manager for current platform.
+    """
+    if is_windows_os():
+        if not is_tool_installed(PackageManager.WINDOWS):
+            raise ValueError('Did not find {0} on the system. \
+                                Please install it or \
+                                try setting install_if_not_found=True'.format(PackageManager.WINDOWS))
+        return PackageManager.WINDOWS
+
+    elif is_mac_os():
+        if not is_tool_installed(PackageManager.MAC):
+            raise ValueError('Did not find {0} on the system. \
+                            Please install it or \
+                            try setting install_if_not_found=True'.format(PackageManager.MAC))
+        return PackageManager.MAC
+
+    elif is_linux_os():
+        if not is_tool_installed(PackageManager.LINUX):
+            raise ValueError('Did not find {0} on the system. \
+                                Please install it or \
+                                try setting install_if_not_found=True'.format(PackageManager.LINUX))
+        return PackageManager.LINUX
+
+    else:
+        raise ValueError('Could not recognize OS')
+
+
+def install_system_packages(package_names):
+    """ Install packages using system package manager.
+    brew for mac, apt for linux, choco for windows
+
+    Args:
+        package_names (:obj:`list` of :obj:`str`): List of packages to install.
+                                                    Eg: ['ninja', 'ccache'].
+    Raises:
+        (ValueError): When either package manager is not found or packages list is empty.
+    """
+    if not package_names:
+        print('No packages specified to install via system pacakge manager')
+        return
+
+    # Get current platform's system package manager
+    package_manager = get_system_package_manager()
+    if not package_manager:
+        raise ValueError('No package manager found on the system. \
+                         Could not install packages {0}'.format(" ".join(package_names)))
+    cmd = [package_manager, 'install'] + package_names
+    if is_linux_os():
+        run_command(cmd, as_root_user=True)
+    else:
+        run_command(cmd)
+
+
+def install_python_packages():
+    """ Install python packages needed as dependencies for cpp sdk build.
+    """
```

Comment:
Nit: For single line docstrings, the terminating """ should be on the same line.